### DISCUSSION TITLE
Can fzf go to the matched item's next item?

### DISCUSSION THREAD

**Discussion by GHolk** [junegunn/fzf#4117] [Q&A] [answered]
For example:

```
cat <<EOF | fzf --reverse
name fb
type social

name rd
type forum

EOF
```

In this case, if I type `rd`, fzf will bring me `name rd`.
Can I go to its next line `type forum` after fzf bring me the line `name rd`?

**Comment by LangLangBart** [comment]
One idea is to process the input by replacing two consecutive newline characters
with a `null` character and setting the `delimiter` to a newline character. Upon
pressing <kbd>Enter</kbd>, the second field of the selected entry will be
printed using the `become`[^1] action.


```bash
cat <<EOF | perl -0777 -pe 's/\n{2,}/\0/gm' | fzf --reverse --read0 --delimiter '\n' --bind 'enter:become:echo {2}'
name fb
type social

name rd
type forum

EOF
```

> [!NOTE]
> The `--read0` flag has been available since version `0.15.0`. The ability to
> display entries with newlines across multiple lines was introduced in
> `0.53.0`.
> <img src="https://github.com/user-attachments/assets/2200688b-0c85-4f52-bf4e-09fe97d9a3d3" width="400">



[^1]: [CHANGELOG.md 0.38.0](https://github.com/junegunn/fzf/blob/master/CHANGELOG.md#0380)

**Reply by GHolk** [reply]
Thanks, but not what I want.

**Reply by LangLangBart** [reply, marked as answer]
> Thanks, but not what I want.

Attempt 2. Upon pressing <kbd>Space</kbd>, the `result` event is activated, the query is
cleared, which triggers the `result` event and you move down.

```bash
cat <<EOF | fzf --reverse --track --bind 'space:rebind(result)+clear-query' --bind 'start:unbind:result' --bind 'result:down+unbind(result)'
name fb
type social

name rd
type forum

EOF
```

**EDIT1**: `--reverse` flag added

**Reply by GHolk** [reply]
Thanks! `--track` is what I want!
I download the latest fzf version then get the `--track` option.

I am not sure how does the result event and the rebind work, but the track option is what I need.
I use this instead of the result event:

```bash
fzf --bind 'alt-enter:execute(echo {};sleep 0.1)+clear-query+refresh-preview'
```

Maybe I will look into the result event later.

**Reply by LangLangBart** [reply]
> but the track option is what I need.

I see.

There is also a `track-current` action (added in `0.40.0` as `track`, and
renamed to `track-current` in `0.49.0`[^1]).
- Ref: https://github.com/junegunn/fzf/discussions/3443

An alternative implementation to yours, without flashing (`execute`) and the
ability to select and unselect (`toggle`) before leaving `fzf`, outcome should
be the same. `{+f}` is a `fzf` placeholder for a temporary file listing all your
selections. 

```bash
cat <<EOF | fzf --reverse --multi --bind 'tab:toggle+track-current+clear-query+refresh-preview' --bind 'enter:become:cat {+f}'
name fb
type social

name rd
type forum

EOF
```

---

> I am not sure how does the result event and the rebind work,

1. Disable the `result` event to avoid triggering all the time.

`--bind 'start:unbind:result'`

```
start
  Triggered only once when fzf finder starts. [...]
```

2. Activate the `result` event when the `Space` key is pressed. Clearing the
   query updates the list and triggers the `result` event.

`--bind 'space:rebind(result)+clear-query'`

3. Assign the `result` event with an action to move one line `down`, and unbinds
   itself:

`--bind 'result:down+unbind:result'`

```
result
  Triggered when the filtering for the current query is complete
  and the result list is ready.
```

[^1]: [fzf/CHANGELOG.md at master · junegunn/fzf · GitHub](https://github.com/junegunn/fzf/blob/master/CHANGELOG.md#0490)

**Reply by GHolk** [reply]
finally I use the tab:
`fzf --layout=reverse --tiebreak=begin --bind=tab:track-current+clear-query,alt-a:top,alt-a:first,alt-e:last`

**Comment by junegunn** [comment]
See #4534
